topdown/http_test.go

```diff
@@ -431,6 +431,130 @@ func TestHTTPRedirectEnable(t *testing.T) {
 	runTopDownTestCase(t, data, "http.send", rule, resultObj.String())
 }
 
+func TestHTTPSendCaching(t *testing.T) {
+	// test server
+	nextResponse := "{}"
+	var requests []*http.Request
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		requests = append(requests, r)
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(nextResponse))
+	}))
+	defer ts.Close()
+
+	// expected result
+
+	var body []interface{}
+	bodyMap := map[string]string{"id": "1", "firstname": "John"}
+	body = append(body, bodyMap)
+
+	// run the test
+	tests := []struct {
```

Comment:
I can tell from the implementation that it's going to do the right thing, but having a test case for multiple POSTs with the same URL and different message bodies would be good since that was the original case that uncovered the bug.

Reply:
I was confused at first, being super sure I added that case.. and I did... just had it doing "GET" with a payload 🤦‍♂ Switching it to POST and we're good now.